RecoverCRPVM/RecoverCRPVM.ps1

```diff
@@ -0,0 +1,175 @@
+    return
+}
+Write-Log "Successfully got the VM Object info for ==> $($vm.Name)" -Color Green
+
+if ($vm.StorageProfile.OsDisk.OsType -eq "Windows") {$windowsVM= $true} else {$windowsVM= $false}
+
+$Vmname = $vm.Name
+$rescueVMNname = "$prefix$Vmname"
+$RescueResourceGroup = "$prefix$ResourceGroup"
+
+#Step 2 Get the Rescue VM Object
+Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$RescueResourceGroup `" -Name `"$rescueVMNname`"" -Color Yellow
+$rescuevm = Get-AzureRmVM -ResourceGroupName $RescueResourceGroup -Name $rescueVMNname
+if (-not $rescuevm)
+{
+    Write-Log "Unable to find the VM `"$rescueVMNname`",  cannot proceed, please verify the VM name and the resource group name." -Color Red
+    return
+}
+Write-Log "Successfully got the VM Object info for the Rescue VM ==>  $rescueVMNname" -Color Green
+
+#Step 3 -Removing the DataDisk from Rescue VM
+$VHDNameShort = ($FixedOsDiskUri.Split('/')[-1]).split('.')[0]
+Write-Log "Removing the Data disk from the Rescue VM ==>  $rescueVMNname" -Color Yellow
+Remove-AzureRmVMDataDisk -VM $rescuevm -Name $VHDNameShort
+Update-AzureRmVM -ResourceGroupName $RescueResourceGroup -VM $rescuevm
+Write-Log "Successfully removed the Data disk from the Rescue VM ==>  $rescueVMNname" -Color Green
+
+#Stop the VM before performing the disk swap.
+Write-Log "Stopping the VM ==> $VmName"  -Color Yellow
+$stopped= Stop-AzureRmVM -ResourceGroupName $ResourceGroup -Name $VmName
+if ($Stopped)
+{
+   Write-Log "Successfully stopped the  VM ==> $VmName" -Color Green
+}
+
+
+#Step 4 -Disk Swapping the OS Disk to point to the fixed OSDisk Uri
+
+Write-Log "Disk Swapping the OS Disk, to point to the fixed OS Disk for VM ==>  $VmName" -Color Yellow
+#before setting the uri, ensure to remove any new line characters
+$FixedOsDiskUri  = $FixedOsDiskUri.Replace("`r`n","")
+$vm.StorageProfile.OsDisk.Vhd.Uri = $FixedOsDiskUri 
+Update-AzureRmVM -ResourceGroupName $ResourceGroup -VM $vm 
+Write-Log "Successfully Disk Swapped the OS Disk,  for VM ==>  $VmName" -Color Yellow
+
+#Step 5 -Start the VM
+Write-Log "Starting the VM ==> $VmName"  -Color Yellow
+$started= Start-AzureRmVM -ResourceGroupName $ResourceGroup -Name $VmName
+if ($Started)
+{
+   Write-Log "Successfully started the  VM ==> $VmName" -Color Green
+}
+
+#Step 5 Open a RDP Connection to the VM
+if ($windowsVM)
+{
+    Write-log "Opening the RDP file to connect to the  VM ==> $VmName" 
+    Get-AzureRmRemoteDesktopFile -ResourceGroupName $ResourceGroup -Name $VmName -Launch
+}
+
+Write-Host "`nWere you able to successfully recover the VM ==> VmName and are you ready to delete all the rescue reources that were created under the Resource Group $RescueResourceGroup (Y/N)?" -ForegroundColor Yellow
+if ((read-host) -eq 'Y')
+{
+    Write-Log "Acknowledged deleting the rescource group ==> $RescueResourceGroup" -color Cyan
+    Remove-AzureRmResourceGroup -Name $RescueResourceGroup -Force
+}
+else
```

Comment:
Doubt: Not sure if we want to keep the log file around? Seems like better to attach to SR and delete?

Reply:
ie. I am proposing it can be done manually instead of in the script

---
In reply to: [160099070](https://github.com/sujasd/azure-support-scripts/pull/1#discussion_r160099070) [](ancestors = 160099070)